fw/cache.c

```diff
@@ -2081,7 +2081,6 @@ tfw_cache_add_body_page(TfwMsgIter *it, char *p, int sz, TfwFrameHdr *frame_hdr,
 
 	++it->frag;
 	skb_fill_page_desc(it->skb, it->frag, page, off, sz);
-	skb_frag_ref(it->skb, it->frag);
```

Comment:
It seems we don't need this reference counter because we just allocated the page with `alloc_page()` and the page already has refcounter =1. However, this code works also for non HTTP/2 branch, were we just get the page from `p` and it seems we need the refcounter. Or need a good explanation what was wrong with HTTP/1 code as well.

A good commit message explaining the problem and how it is fixed and why the fix is correct, would help.

Reply:
@krizhanovsky @s0nx We can't remove `skb_frag_ref(it->skb, it->frag);` for h1. Because fragment uses same page as TBD, unrefing the fragment will cause TDB memory corruption.